build-data-capturing-gradle-samples/capture-slow-workunit-executions/gradle-slow-task-executions.gradle.kts

```diff
@@ -0,0 +1,36 @@
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension
+import com.gradle.scan.plugin.BuildScanExtension
+
+/**
+ * This Gradle script captures all tasks of a given type taking longer to execute than a certain threshold,
+ * and adds these as custom values.
+ */
+
+project.extensions.configure<GradleEnterpriseExtension>() {
+    buildScan {
+        val THRESHOLD_MILLIS = 15 * 60 * 1000 // 15 min
+        val api = buildScan
+        allprojects {
+            tasks.withType<JavaCompile> {
```

Comment:
I think in the Kotlin DSL, the behavior of `withType` is different and one must call `.configureEach` to avoid eager configuration, can you double check?

Reply:
🤯 You're right, that is documented as well